pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -340,6 +373,36 @@ func (mi *StorageInitializerInjector) InjectStorageInitializer(pod *v1.Pod) erro
 	return nil
 }
 
+// Use values from crd container spec from most fields. Use default values from initContainer for some fields.
+func mergeContainerSpecs(initContainer *v1.Container, crdSpec *v1.Container) (*v1.Container, error) {
```

Comment:
can we do strategy merge with the two container spec? I'd expect we do not need to explicitly enumerate all the fields and override here

Reply:
Updated it to use strategy merge.